github_actions/lib/dependabot/github_actions/file_parser.rb

```diff
@@ -55,8 +55,20 @@ def workfile_file_dependencies(file)
       def build_github_dependency(file, string)
         details = string.match(GITHUB_REPO_REFERENCE).named_captures
         name = "#{details.fetch('owner')}/#{details.fetch('repo')}"
```

Comment:
The above two lines are duplicated in the `github_dependency` method. I think you can remove them by changing `github_dependency`'s third parameter to receive a hostname, instead of a URI, and building the URI inside the `github_dependency` method, right when instantiating the `Dependency`:

```ruby
Dependency.new(
  # ...
  url: "https://#{hostname}/#{name}"
  # ...
)
```